ocaml/xapi/helpers.ml

```diff
@@ -1907,7 +1907,10 @@ end = struct
 
   let systemctl_ cmd = systemctl cmd |> ignore
 
-  let reload () = systemctl_ "reload-or-restart"
+  let reload ?(wait = 5.0) () =
+    systemctl_ "reload-or-restart" ;
+    (* We can't be sure that the reload is finished, so wait a moment *)
+    Unix.sleepf wait
```

Comment:
This releases the runtime lock and should allow other threads to run, so should be fine.

Reply:
We normally use `Thread.delay`. I don't know if `sleep` is nowadays suitable for threaded programs, but it wasn't in the past.